File: Sound/Application/Services/SoundService.cs
using Application.Repositories.IRepositories;
using Application.Services.IServices;
using Data.Dto;
using Data.Dto.Mix;
using Data.Common;

namespace Application.Services
{
    public class SoundService : ISoundService
    {
        private ISoundRepo _soundRepo;
        public SoundService(ISoundRepo soundRepo)
        {
            _soundRepo = soundRepo;
        }
        public async Task<ResponseData<Pagination<SoundDto>>> GetSound(int PageSize = 10, int PageNumber = 1)
        {
            return await _soundRepo.GetSound(PageSize, PageNumber);
        }
        public async Task<ResponseData<string>> AddSound(AddSoundDto sound, FileSound file)
        {
            return await _soundRepo.AddSound(sound, file);
        }
        public async Task<ResponseData<string>> UpdateSound(EditSoundDto sound, FileSound file)
        {
            return await _soundRepo.UpdateSound(sound, file);
        }
        public async Task<ResponseData<string>> DeleteSound(long id)
        {
            return await _soundRepo.DeleteSound(id);
        }
        public async Task<ResponseData<List<GetMixSoundDto>>> GetSound(int idMix)
        {
            return await _soundRepo.GetSound(idMix);
        }
        public async Task<ResponseData<string>> CreateMix(CreateMixSoundDto mix)
        {
            return await _soundRepo.CreateMix(mix);
        }
        public async Task<ResponseData<string>> SaveMix(UpdateMixSoundDto update)
        {
            return await _soundRepo.SaveMix(update);
        }
        public async Task<ResponseData<string>> ActiveSound(long id)
        {
            return await _soundRepo.ActiveSound(id);
        }
    }
}

File: Fe_Admin/fe_sound/src/services/sound/sound.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../share/Environment/environment';
import { SoundDto, AdminSoundDto, GetMixSoundDto, CreateMixSoundDto, UpdateMixSoundDto, AddSound, EditSound } from './sound.dtos';
import { ResponseData, Pagination, GetList } from '../../share/Dtos/Dtos.Share';
import { toFormBody, toHttpParams } from '../../share/Services/Service.Share';
import { api } from '../../share/Environment/api.link';
@Injectable({
    providedIn: 'root'
})
export class SoundService {
    constructor(private http: HttpClient) { }

    getSound(dataGet: GetList): Observable<ResponseData<Pagination<SoundDto>>> {
        return this.http.get<ResponseData<Pagination<SoundDto>>>(api.sound.getSound, { params: toHttpParams(dataGet) });
    }

    getSoundByAdmin(dataGet: GetList): Observable<ResponseData<Pagination<AdminSoundDto>>> {
        return this.http.get<ResponseData<Pagination<AdminSoundDto>>>(api.sound.getSoundByAdmin, { params: toHttpParams(dataGet) });
    }

    addSound(formData: AddSound): Observable<ResponseData<string>> {
        return this.http.post<ResponseData<string>>(api.sound.addSound, toFormBody(formData));
    }

    updateSound(formData: EditSound): Observable<ResponseData<string>> {
        return this.http.put<ResponseData<string>>(api.sound.updateSound, formData);
    }

    deleteSound(idSound: number): Observable<ResponseData<string>> {
        return this.http.delete<ResponseData<string>>(api.sound.deleteSound, { params: toHttpParams({ id: idSound }) });
    }

    activateSound(idSound: number): Observable<ResponseData<string>> {
        return this.http.patch<ResponseData<string>>(api.sound.activateSound, null, {
            params: toHttpParams({ id: idSound })
        });
    }

    getSoundMix(idMix: number): Observable<ResponseData<GetMixSoundDto[]>> {
        return this.http.get<ResponseData<GetMixSoundDto[]>>(api.sound.getSoundMix, {
            params: toHttpParams({ idMix: idMix })
        });
    }

    createMix(mixData: CreateMixSoundDto): Observable<ResponseData<string>> {
        return this.http.post<ResponseData<string>>(api.sound.createMix, mixData);
    }

    saveMix(updateData: UpdateMixSoundDto): Observable<ResponseData<string>> {
        return this.http.put<ResponseData<string>>(api.sound.saveMix, updateData);
    }
}
